src/NServiceBus.Transport.Msmq/MessagePump.cs

```diff
@@ -171,29 +181,38 @@ async Task InnerProcessMessages()
 
         Task ReceiveMessage()
         {
-            return TaskEx.Run(async state =>
+            // We pass a fake state to make sure we benefit from lamda delegate caching. See https://github.com/Particular/NServiceBus/issues/3884
```

Comment:
```suggestion
            // We pass a state to make sure we benefit from lamda delegate caching. See https://github.com/Particular/NServiceBus/issues/3884
```

it's not really a "fake"?